asset.js

```diff
@@ -0,0 +1,73 @@
+var get = require('simple-get')
+var noop = Object.assign({
+  http: function () {},
+  silly: function () {}
+}, require('noop-logger'))
+var util = require('./util')
+var url = require('url')
+var tunnel = require('tunnel-agent')
+
+function findAssetId (opts, cb) {
+  if (!opts.token) return cb()
```

Comment:
I'd prefer if we could move this outside the call, it's basically a "hidden" if-statement that also is calling back syncronously. E.g. first check if `opts.token` is set and then call `findAssetid(opts, cb)`.